src/main/java/gyro/google/compute/BackendBucketCdnPolicy.java

```diff
@@ -69,10 +244,63 @@ public String primaryKey() {
+            BackendBucketCdnCacheKeyPolicy cacheKeyPolicy = newSubresource(BackendBucketCdnCacheKeyPolicy.class);
+            cacheKeyPolicy.copyFrom(backendBucketCdnPolicy.getCacheKeyPolicy());
+            setCacheKeyPolicy(cacheKeyPolicy);
+        }
+
+        setBypassCacheOnRequestHeader(null);
+        if (!backendBucketCdnPolicy.getBypassCacheOnRequestHeadersList().isEmpty()) {
+            setBypassCacheOnRequestHeader(backendBucketCdnPolicy.getBypassCacheOnRequestHeadersList().stream()
+                .map(o -> {
+                    BackendBucketCdnBypassCacheOnRequestHeader bypassCacheOnRequestHeader = newSubresource(BackendBucketCdnBypassCacheOnRequestHeader.class);
+                    bypassCacheOnRequestHeader.copyFrom(o);
+                    return bypassCacheOnRequestHeader;
+                })
+                .collect(Collectors.toList()));
+        }
+
+        setNegativeCaching(backendBucketCdnPolicy.getNegativeCaching());
+        setNegativeCachingPolicy(null);
+        if (!backendBucketCdnPolicy.getNegativeCachingPolicyList().isEmpty()) {
+            setNegativeCachingPolicy(backendBucketCdnPolicy.getNegativeCachingPolicyList().stream()
+                .map(o -> {
+                    BackendBucketCdnNegativeCachingPolicy negativeCachingPolicy = newSubresource(BackendBucketCdnNegativeCachingPolicy.class);
+                    negativeCachingPolicy.copyFrom(o);
+                    return negativeCachingPolicy;
+                })
+                .collect(Collectors.toList()));
+        }
+
+        setClientTtl(backendBucketCdnPolicy.getClientTtl());
+        setDefaultTtl(backendBucketCdnPolicy.getDefaultTtl());
+        setMaxTtl(backendBucketCdnPolicy.getMaxTtl());
+        setRequestCoalescing(backendBucketCdnPolicy.getRequestCoalescing());
+        setServeWhileStale(backendBucketCdnPolicy.getServeWhileStale());
     }
 
     com.google.cloud.compute.v1.BackendBucketCdnPolicy toBackendBucketCdnPolicy() {
         return com.google.cloud.compute.v1.BackendBucketCdnPolicy.newBuilder()
-            .setSignedUrlCacheMaxAgeSec(getSignedUrlMaxAge()).build();
+            .setSignedUrlCacheMaxAgeSec(getSignedUrlMaxAge())
+            .setCacheMode(getCacheMode())
+            .setCacheKeyPolicy(getCacheKeyPolicy() != null ? getCacheKeyPolicy().toBackendBucketCdnPolicyCacheKeyPolicy() : new BackendBucketCdnCacheKeyPolicy().toBackendBucketCdnPolicyCacheKeyPolicy())
+            .addAllBypassCacheOnRequestHeaders(getBypassCacheOnRequestHeader().stream()
+                .map(BackendBucketCdnBypassCacheOnRequestHeader::toBackendBucketCdnPolicyBypassCacheOnRequestHeader)
+                .collect(Collectors.toList()))
+            .setClientTtl(getClientTtl())
+            .setDefaultTtl(getDefaultTtl())
+            .setMaxTtl(getMaxTtl())
+            .setNegativeCaching(getNegativeCaching())
+            .addAllNegativeCachingPolicy(getNegativeCachingPolicy().stream()
```

Comment:
I had the `negativeCaching` set to `true` with a `negativeCachingPolicy` set. then, I tried to remove the policy and set `negativeCaching` back to `false` which resulted in the following error:

```
Caused by: Error: Invalid value for field 'resource.cdnPolicy.negativeCaching': 'false'. negative_caching must be enabled to config the negative_caching_policy.
```